test/performance/kperf_helper.go

```diff
@@ -0,0 +1,154 @@
+//go:build performance
+// +build performance
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	networkingv1alpha1 "knative.dev/networking/pkg/client/clientset/versioned/typed/networking/v1alpha1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/logging"
+	"knative.dev/serving/pkg/client/clientset/versioned"
+	autoscalingv1alpha1 "knative.dev/serving/pkg/client/clientset/versioned/typed/autoscaling/v1alpha1"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+	"knative.dev/serving/test"
+)
+
+// Setup creates client to run Knative Service requests
+func testSetup(ctx context.Context, t testing.TB, namespace ...string) *test.Clients {
+	t.Helper()
+	logging.InitializeLogger()
+
+	cfg, err := pkgTest.Flags.GetRESTConfig()
+	if err != nil {
+		t.Fatal("Couldn't get REST config", "error", err)
+	}
+
+	ctx, informers := injection.Default.SetupInformers(ctx, cfg)
+	if err := controller.StartInformers(ctx.Done(), informers...); err != nil {
+		t.Fatal("Couldn't initialize informers", "error", err)
+	}
+
+	clients, err := test.NewClients(cfg, "default")
+	if err != nil {
+		t.Fatal("Couldn't initialize clients", "error", err)
+	}
+	return clients
+}
+
+func getResultJson(tmpFolder string, result interface{}) error {
+	d, err := os.Open(tmpFolder)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	files, err := d.Readdir(-1)
+	if err != nil {
+		return err
+	}
+
+	var jsonFile []byte
+	for _, file := range files {
+		if file.Mode().IsRegular() {
+			if filepath.Ext(file.Name()) == ".json" {
+				f, err := os.Open(fmt.Sprintf("%s/%s", tmpFolder, file.Name()))
+				jsonFile, _ = ioutil.ReadAll(f)
+				if err != nil {
+					return err
+				}
+				break
+			}
+		}
+	}
+
+	json.Unmarshal([]byte(jsonFile), &result)
+	return nil
+}
+
+func servingClientForNamespace(namespace string) (*test.ServingClients, error) {
```

Comment:
Related to my earlier comment. This could probably just return ServingClient from the test.Clients struct.